source/ethereum-clients/cpp-ethereum/automation.rst

```diff
@@ -0,0 +1,88 @@
+.. _cpp-ethereum-automation:
+
+################################################################################
+Automation setup for cpp-ethereum
+################################################################################
+
+At the time of writing all of the automation for cpp-ethereum is driven using
+a Jenkins instance hosted at http://52.28.164.97.
+
+There is a "pretty alias" to the setup at http://ethbuilds.com, but that is
+owned by
+`Bob Summerwill <http://bobsummerwill.com/about>`_ personally, not by the
+:ref:`Ethereum Foundation <foundation>`, and might end up pointing to something else at a later date.
+
+..  image:: ../../img/jenkins.png
+    :height: 256px
+    :width: 256px
+
+
+It runs in parallel with the `Ethereum buildbot instance <https://builds.ethereum.org/>`_
+which is used for Go and Python builds.
+
+The fact that we have two different automation systems is not ideal, and is for
+historical reasons.   It would make sense for us to consolidate all
+:ref:`Ethereum Foundation <foundation>` projects into a single, consistent
+automation setup, but that is a non-trivial amount of work.   We're talking about
+that.  It will be much easier to assess that when we have completed the
+`repo reorg <https://github.com/ethereum/webthree-umbrella/issues/251>`_
+for the C++ codebase.
+
+The current Jenkins setup is `missing a canonical continuous integration target <https://github.com/ethereum/webthree-umbrella/issues/247>`_,
```

Comment:
No - even that change wouldn't be sufficient for two reasons.

1. By canonical, I'm talking about post-commit automated builds and testing.    Every project I have ever worked on has had such an automation target (and often ONLY that target), because it verifies what you actually have in the repository, irrespective of how it got there.     We don't have a timeline showing the state of the build at every commit for our umbrella (which would shortly be replaced by by N of those, one for each repo).     It's great that we're testing the PRs, but that doesn't guarantee that people don't commit them when they show as broken.      Right now we are blind to whether the build *is* broken.    We are working on the flawed inference that we will never break it because our the gates we have in place.

2. Also, just adding that specific gate only tests on Linux.     And it only tests sub-module updates which are being done through it.     And it won't even make sense after the repo reorg anyway.